modin/pandas/io.py

```diff
@@ -63,12 +65,37 @@
 from pandas.io.parsers import TextFileReader
 from pandas.io.parsers.readers import _c_parser_defaults
 
+from modin.config import ExperimentalNumPyAPI
 from modin.error_message import ErrorMessage
 from modin.logging import ClassLogger, enable_logging
-from modin.utils import _inherit_docstrings, expanduser_path_arg
+from modin.utils import (
+    SupportsPrivateToNumPy,
+    SupportsPrivateToPandas,
+    SupportsPublicToNumPy,
+    _inherit_docstrings,
+    expanduser_path_arg,
+)
+
+# below logic is to handle circular imports without errors
+if TYPE_CHECKING:
+    from .dataframe import DataFrame
+    from .series import Series
+
+
+class ModinObjects:
+    """Lazily import Modin classes and provide an access to them."""
+
+    _dataframe = None
+
+    @classmethod
+    @property
```

Comment:
Let's not use them together: https://stackoverflow.com/questions/128573/using-property-on-classmethods/64738850#64738850

Reply:
changed to using this as a classmethod-constructor